docs/examples/multi-agent-workflow.md

```diff
@@ -0,0 +1,233 @@
+In this example we implement a multi-agent Workflow we have a single "Researcher" Agent that conducts research on a topic, and then fans out to multiple "Writer" Agents to write blog posts based on the research.
+
+By splitting up our workloads across multiple Structures, we can parallelize the work and leverage the strengths of each Agent. The Researcher can focus on gathering data and insights, while the Writers can focus on crafting engaging narratives.
+Additionally, this architecture opens us up to using services such as [Griptape Cloud](https://www.griptape.ai/cloud) to have each Agent run on a separate machine, allowing us to scale our Workflow as needed.
+
+
+```python
+import json
+import os
+from pathlib import Path
+
+from griptape.drivers import WebhookEventListenerDriver
+from griptape.events import EventListener, FinishStructureRunEvent
+from griptape.rules import Rule, Ruleset
+from griptape.structures import Agent, Workflow
+from griptape.tasks import CodeExecutionTask, StructureRunTask
+from griptape.tools import (
+    TaskMemoryClient,
+    WebScraper,
+    WebSearch,
+)
+
+
+def build_researcher():
+    """Builds a Researcher Structure."""
+    researcher = Agent(
+        id="researcher",
+        tools=[
+            WebSearch(
+                google_api_key=os.environ["GOOGLE_API_KEY"],
+                google_api_search_id=os.environ["GOOGLE_API_SEARCH_ID"],
+                off_prompt=False,
```

Comment:
> Is it because the web search results are expected to always be short enough to fit in the prompt
Yes, exactly that.
> If that is this only reason, then the example should still work when setting it to True, right?
Yes, but there would be an extra step introduced by putting it into Task Memory and then needing to retrieve it.

Have you read [this doc](https://github.com/griptape-ai/griptape/pull/757/files#diff-663d63822a27214e8158beb30e242975468127845bcf5bdea719e4095d41334b) yet? Hoping to clear up some of this confusion around Task Memory.

Reply:
Oh yeah I actually started looking at that PR after this one. That default makes a pretty big difference actually